gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+          && (!Character.isLetter(c) || !Character.isLowerCase(c))) {
+        return false;
+      }
+    }
+    if (value.length() > 0 && (!Character.isLetter(value.charAt(0)) || value.endsWith("-"))) {
+      return false;
+    }
+    return value.length() >= minLength && value.length() <= maxLength;
+  }
+
+  Response create(Project project) {
+    project.setLifecycleState("ACTIVE");
+    project.setProjectNumber(Math.abs(PROJECT_NUMBER_GENERATOR.nextLong() % Long.MAX_VALUE));
+    project.setCreateTime(ISODateTimeFormat.dateTime().print(System.currentTimeMillis()));
+    Response response;
+    String customErrorMessage = checkForProjectErrors(project);
+    if (customErrorMessage != null) {
+      response = Error.INVALID_ARGUMENT.response(customErrorMessage);
+    } else if (projects.containsKey(project.getProjectId())) {
+      response = Error.ALREADY_EXISTS.response(
+          "A project with the same project ID (" + project.getProjectId() + ") already exists.");
+    } else {
+      projects.put(project.getProjectId(), project);
+      try {
+        String createdProjectStr = jsonFactory.toString(project);
+        response = new Response(HttpURLConnection.HTTP_OK, createdProjectStr);
+      } catch (IOException e) {
+        response =
+            Error.INTERNAL_ERROR.response("Error serializing project " + project.getProjectId());
+      }
+    }
+    return response;
+  }
+
+  Response delete(String projectId) {
+    Project project = projects.get(checkNotNull(projectId));
+    Response response;
+    if (project == null) {
+      // when possible, change this to 404 (#440)
+      response = Error.PERMISSION_DENIED.response(
+          "Error when deleting " + projectId + " because the project was not found.");
+    } else if (!project.getLifecycleState().equals("ACTIVE")) {
+      response = Error.FAILED_PRECONDITION.response(
+          "Error when deleting " + projectId + " because the lifecycle state was not ACTIVE.");
+    } else {
+      project.setLifecycleState("DELETE_REQUESTED");
+      response = new Response(HttpURLConnection.HTTP_OK, "{}");
+    }
+    return response;
+  }
+
+  Response get(String projectId, String[] fields) {
+    if (!projects.containsKey(checkNotNull(projectId))) {
+      // when possible, change this to 404 (#440)
+      return Error.PERMISSION_DENIED.response("Project " + projectId + " not found.");
+    }
+    Project project = projects.get(projectId);
+    try {
+      return new Response(
+          HttpURLConnection.HTTP_OK, jsonFactory.toString(extractFields(project, fields)));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          "Error when serializing project " + project.getProjectId());
+    }
+  }
+
+  Response list(final Map<String, Object> options) {
+    // Use pageSize and pageToken options when Cloud Resource Manager does so (#421)
+    List<String> projectsSerialized = new ArrayList<>();
+    for (Project p : projects.values()) {
```

Comment:
Yes if they're in `DELETE_REQUESTED` state and probably yes `DELETE_IN_PROGRESS`.  The latter is hard to test because I only got notifications that my projects were going to be deleted (after calling delete() to put them in the `DELETE_REQUESTED` state) a couple days after deleting them, so it's hard to reproduce an instance and check it at the right time.